Enterprise-CMCS/eAPD
# 4624 Navigating between activities using the nav menu doesn't update the Activity Name

Resolves #4624 

### Description
This fix displays the correct activity name (on Activity Overview page) when navigating between activities. Before the fix, the previously viewed activity name was persisting in the text box under "Activity name".

### Significant changes or possible side effects
- Added a `useEffect` function to the `NameForm` component which listens for changes in the "name" value and updates the field in reach hooks form when there is a change (in this case when we navigate from one activity to another)

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| At least two activities exist (which use the NameForm component) | Navigating between the two activities | The Activity name text box should display the correct Activity's name | cypress |

### Steps to manually verify this change

1. Navigate to an APD that has multiple activities (at least 2 for MMIS or at least 3 for HITECH)
2. Navigate between the Activity Overview page of the two activities that have an Activity name textbox
3. Ensure that the name inside the textbox correctly reflects the activity that is selected

### This pull request is ready to code review when

      tests, refactor
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when


### This pull request is ready to review when QA has


### This pull request can be merged when
